system-addon/lib/SectionsManager.jsm

```diff
@@ -278,11 +278,16 @@ class SectionsFeed {
       case at.PREFS_INITIAL_VALUES:
         SectionsManager.init(action.data);
         break;
-      case at.PREF_CHANGED:
-        if (action.data && action.data.name.match(/^feeds.section.(\S+).options$/i)) {
-          SectionsManager.addBuiltInSection(action.data.name.slice(0, -8), action.data.value);
+      case at.PREF_CHANGED: {
+        if (action.data) {
+          const matched = action.data.name.match(/^feeds.section.(\S+).options$/i);
+          if (matched) {
+            SectionsManager.addBuiltInSection(action.data.name.slice(0, -8), action.data.value);
```

Comment:
Seems like if we're touching this, might as well grab the feed pref name from the regex instead of slicing some number.
`/^(feeds\.section\.(.+))\.options$/i`

Reply:
True, I'll take care of that.